File: imobil-project/bin/imobil.dart
import 'dart:core';
import 'Poza.dart';
import 'Proprietar.dart';
import 'Agent.dart';
import 'Client.dart';
class Imobil
{
  Proprietar proprietar;
  String locatie;
  int dimensiune;
  Poza poze;
  Agent agent;
  Client client_potential;
  Imobil(proprietar, String locatie, int dimensiune, Poza poze)
  {
    this.proprietar = proprietar;
    this.locatie = locatie;
    this.dimensiune = dimensiune;
    this.poze = poze;
  }
  Proprietar get PROPRIETAR => proprietar;
  set PROPRIETAR(Proprietar value) => proprietar=value;

  String get LOCATIE => locatie;
  set LOCATIE(String value) => locatie=value;

  int get DIMENSIUNE=>dimensiune;
  set DIMENSIUNE(int value)=>dimensiune=value;

  Poza get POZE=>poze;
  set POZE(Poza value)=>poze=value;

  Agent get Asignare_Agent=>agent;
  set Asignare_Agent(Agent value)=>agent=value;

  Client get Asignare_Client_Potential=>client_potential;
  set Asignare_Client_Potential(Client value)=>client_potential=value;

  double OfertaVSPret()
  {
    return 100-((100 * client_potential.Buget) / proprietar.pretcerut);
  }
  @override
  String toString() {
    StringBuffer sb = new StringBuffer();
    sb.writeln('~~IMOBIL~~');
    sb.writeln('PROPRIETAR: ' +proprietar.toString());
    sb.writeln('LOCATIE: '+locatie.toString());
    sb.writeln('DIMENSIUNE: '+dimensiune.toString());
    sb.writeln('POZE: '+poze.toString());
    return sb.toString();

  }
}

File: Imobil C#/Imobil/Imobil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Imobil
{
    class Imobil
    {
        public Proprietar proprietar;
        public string locatie;
        public int dimensiune;
        public Poza poze;
        public Agent agent;
        public Client clientpotential; 

        public Imobil(Proprietar proprietar,string locatie,int dimensiune,Poza poze)
        {
            this.proprietar = proprietar;
            this.locatie = locatie;
            this.dimensiune = dimensiune;
            this.poze = poze;
        }
        public Proprietar PROPRIETAR
        {
            get
            {
                return proprietar;
            }
            set
            {
                proprietar = value;
            }
        }
        public string Locatie
        {
            get
            {
                return locatie;
            }
            set
            {
                locatie = value;
            }
        }
        public int Dimensiune
        {
            get
            {
                return dimensiune;
            }
            set
            {
                dimensiune = value;
            }
        }
        public Poza Poze
        {
            get
            {
                return poze;
            }
            set
            {
                poze = value;
            }
        }
        public Agent AsignareAgent
        {
            get
            {
                return agent;
            }
            set
            {
                agent = value;
            }
        }
        public Client AsignareClientPotential
        {
            get
            { 
                return clientpotential;
            }
            set
            {
                clientpotential = value;
            }
        }
        public Client EliminareClientPotential
        {
            set
            {
                clientpotential = null;
            }
        }
        public float OfertaVSPret()
        {
            return 100-((100 * clientpotential.Buget) / proprietar.pretcerut); 
        }
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendLine("~~IMOBIL~~");
            sb.AppendLine($"PROPRIETAR: {proprietar}");
            sb.AppendLine($"LOCATIE: {locatie}");
            sb.AppendLine($"DIMENSIUNE: {dimensiune}");
            sb.AppendLine($"POZE: {poze}");
            return sb.ToString();        
        }
    }
}
